src/components/SignupFormContainer.js

```diff
@@ -69,34 +211,142 @@ class SignupFormContainer extends React.Component {
+			this.setState({sogIdentity: uniq}, function () {
+				body['sogIdentity'] = this.state.sogIdentity;
+				updateUser(userData, body)
+					.then((data) => {
+						this.setState({userData: data.user}, function () {});
+					})
+					.catch((error) => {
+						handleMessageNew(<FormattedMessage id="error.unspecified" />);
+					});
+			});
+		} else {
+			body['sogIdentity'] = this.state.sogIdentity;
+			updateUser(userData, body)
+				.then((data) => {
+					this.setState({userData: data.user}, function () {});
+				})
+				.catch((error) => {
+					handleMessageNew(<FormattedMessage id="error.unspecified" />);
+				});
+		}
+
+		if (this.state.ethnicityRace.includes('aboutyou.answer-other')) {
+			let specifiedEth = 'specifiedEthnicity: ' + this.state.specifiedEthnicity;
+			tempArray = this.state.ethnicityRace.slice();
+			tempArray.push(specifiedEth);
+			let uniq = [...new Set(tempArray)];
+			this.setState({ethnicityRace: uniq}, function () {
+				body['ethnicityRace'] = this.state.ethnicityRace;
+				updateUser(userData, body)
+					.then((data) => {
+						this.setState({userData: data.user}, function () {});
+					})
+					.catch((error) => {
+						handleMessageNew(<FormattedMessage id="error.unspecified" />);
+					});
+			});
+		} else {
+			body['ethnicityRace'] = this.state.ethnicityRace;
+			updateUser(userData, body)
+				.then((data) => {
+					this.setState({userData: data.user}, function () {});
+				})
+				.catch((error) => {
+					handleMessageNew(<FormattedMessage id="error.unspecified" />);
+				});
+		}
+
+		//update other attributes here
+		updateUser(userData, body)
+			.then((data) => {
+				this.setState({userData: data.user}, function () {});
+			})
+			.catch((error) => {
+				handleMessageNew(<FormattedMessage id="error.unspecified" />);
+			});
+
+		//determine next step in the workflow
+		if (this.state.activeStep === 10 || this.state.activeStep === 5) {
+			this.props.handleRequestOpen('thankyou');
+		} else {
+			this.handleStepNext();
 		}
 	}
 
 	handleSignUp(event) {
 		event.preventDefault();
 		const {handleMessageNew, handleRequestClose, handleRequestOpen} =
 			this.props;
-		const {email, name, password, passwordConfirmation, selection} = this.state;
+		const {
+			email,
+			name,
+			password,
+			selection,
+			currentLocation,
+			orgType,
+			specifiedOrgType
+		} = this.state;
+
 		const isProfessional = selection === 'lawyer' || selection === 'provider';
+		const emailTest = new RegExp(/\S+@\S+\.\S+/);
+		const pswdTest = new RegExp(
+			'^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&?])(?=.{10,})'
+		);
 
-		if (password.length < 8) {
-			handleMessageNew(<FormattedMessage id="error.password-length" />);
+		if (pswdTest.test(password) === false) {
+			handleMessageNew(<FormattedMessage id="error.password-format" />);
 			return;
 		}
 
-		if (password !== passwordConfirmation) {
-			handleMessageNew(<FormattedMessage id="error.password-mismatch" />);
+		if (emailTest.test(email) === false) {
```

Comment:
`test()` returns a boolean. you don't need to compare it, just use the returned value to make assertions.
```suggestion
		if (!emailTest.test(email)) {
```